src/dex/weth/weth.ts

```diff
@@ -0,0 +1,223 @@
+{
+  readonly hasConstantPriceLargeAmounts = true;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(WethConfig);
+
+  public static getAddress(network: number = 1): Address {
+    return WethMap[network];
+  }
+
+  logger: Logger;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {},
+    protected unitPrice = BigInt(1e18),
+    protected poolGasCost = WethConfig[dexKey][network].poolGasCost,
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {}
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] || null;
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (srcToken.address.toLowerCase() === destToken.address.toLowerCase()) {
+      return [];
+    }
+
+    const tokenAddress = [
+      srcToken.address.toLowerCase(),
+      destToken.address.toLowerCase(),
+    ]
+      .sort((a, b) => (a > b ? 1 : -1))
+      .join('_');
+
+    const poolIdentifier = `${this.dexKey}_${tokenAddress}`;
+    return [poolIdentifier];
+  }
+
+  // Returns pool prices for amounts.
+  // If limitPools is defined only pools in limitPools
+  // should be used. If limitPools is undefined then
+  // any pools can be used.
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<WethData>> {
+    const isWETHSwap =
+      (isETHAddress(srcToken.address) &&
+        isWETH(destToken.address, this.network)) ||
+      (isWETH(srcToken.address, this.network) &&
+        isETHAddress(destToken.address));
+
+    if (!isWETHSwap) return null;
+
+    return [
+      {
+        prices: amounts,
+        unit: this.unitPrice,
+        gasCost: this.poolGasCost,
+        exchange: this.dexKey,
+        poolAddresses: [Weth.getAddress(this.network)],
+        data: null,
+      },
+    ];
+  }
+
+  // Encode params required by the exchange adapter
+  // Used for multiSwap, buy & megaSwap
+  // Hint: abiCoder.encodeParameter() could be useful
```

Comment:
```suggestion
  // Hint: abiCoder.encodeParameter() could be useful
```

Reply:
Removed all template comments. You are right. They are not needed here